components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -141,173 +162,1014 @@ public class LegacyJobOrchestrator : IJobOrchestrator {
+            if (_publishedNodesProvider != null) {
+                _publishedNodesProvider.EnableRaisingEvents = false;
+                _publishedNodesProvider.Changed -= _fileSystemWatcher_Changed;
+                _publishedNodesProvider.Created -= _fileSystemWatcher_Created;
+                _publishedNodesProvider.Renamed -= _fileSystemWatcher_Renamed;
+                _publishedNodesProvider.Deleted -= _fileSystemWatcher_Deleted;
             }
+            _lockConfig?.Dispose();
+            _lockJobs?.Dispose();
         }
 
         private void _fileSystemWatcher_Changed(object sender, FileSystemEventArgs e) {
-            if (e.ChangeType == WatcherChangeTypes.Deleted) {
-                _logger.Information("Published nodes file deleted, cancelling all publishing jobs");
-                _lock.Wait();
+            _logger.Debug("File {publishedNodesFile} changed. Triggering file refresh ...", _legacyCliModel.PublishedNodesFile);
+            RefreshJobFromFile();
+        }
+
+        private void _fileSystemWatcher_Created(object sender, FileSystemEventArgs e) {
+            _logger.Debug("File {publishedNodesFile} created. Triggering file refresh ...", _legacyCliModel.PublishedNodesFile);
+            RefreshJobFromFile();
+        }
+
+        private void _fileSystemWatcher_Renamed(object sender, FileSystemEventArgs e) {
+            _logger.Debug("File {publishedNodesFile} renamed. Triggering file refresh ...", _legacyCliModel.PublishedNodesFile);
+            RefreshJobFromFile();
+        }
+
+        private void _fileSystemWatcher_Deleted(object sender, FileSystemEventArgs e) {
+            _logger.Debug("File {publishedNodesFile} deleted. Clearing configuration ...", _legacyCliModel.PublishedNodesFile);
+            _lockConfig.Wait();
+            try {
+                _publishedNodesEntries.Clear();
+                _lastKnownFileHash = string.Empty;
+                _lockJobs.Wait();
                 try {
                     _availableJobs.Clear();
                     _assignedJobs.Clear();
-                    _lastKnownFileHash = string.Empty;
+                    _publisherDiagnosticInfo.Clear();
                 }
                 finally {
-                    _lock.Release();
+                    _lockJobs.Release();
                 }
             }
-            else {
-                RefreshJobFromFile();
+            finally {
+                _lockConfig.Release();
             }
         }
 
-        private static string GetChecksum(string file) {
-            using (var stream = File.OpenRead(file)) {
-                var sha = new SHA256Managed();
-                var checksum = sha.ComputeHash(stream);
-                return BitConverter.ToString(checksum).Replace("-", string.Empty);
+        private static string GetChecksum(string content) {
+            if (string.IsNullOrEmpty(content)) {
+                return null;
+            }
+            var sha = new SHA256Managed();
```

Comment:
Check if this is disposable and use using

Reply:
Fixed to use `using`.